incompatibility/incompatibility-report/node-search.go

```diff
@@ -0,0 +1,36 @@
+package incompatibility
+
+import (
+	"errors"
+	"strconv"
+
+	"gopkg.in/yaml.v3"
+)
+
+func findNode(node *yaml.Node, path ...string) (*yaml.Node, error) {
+	if len(path) == 0 {
+		return node, nil
+	}
+	// Sequence Index
+	if node.Kind == yaml.SequenceNode {
+		ind, err := strconv.Atoi(path[0])
+		if err != nil || ind >= len(node.Content) {
+			return nil, errors.New("invalid index parsed")
+		}
+		return findNode(node.Content[ind], path[1:]...)
+	}
+	//Look for key value
+	for i := 0; i < len(node.Content)-1; i += 2 {
+		key := node.Content[i]
+		val := node.Content[i+1]
+		if key.Value != path[0] {
+			continue
+		}
+		if len(path) == 1 {
```

Comment:
discussed over video-chat: would like to see the parts that are important to the loop to be in the loop and the other parts extracted.

Reply:
made helper function